src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -226,60 +231,80 @@ def read_cell_language(self, lines: list[str]) -> CellLanguage | None:
             return None
         return None
 
-    def new_cell(self, source: str) -> Cell:
-        return self._new_cell(source)
+    def new_cell(self, source: str, original_offset: int) -> Cell:
+        return self._new_cell(source, original_offset)
 
     def extract_cells(self, source: str) -> list[Cell] | None:
         lines = source.split('\n')
         header = f"{self.comment_prefix} {NOTEBOOK_HEADER}"
         if not lines[0].startswith(header):
             raise ValueError("Not a Databricks notebook source!")
 
-        def make_cell(cell_lines: list[str]):
+        def make_cell(cell_lines: list[str], start: int):
             # trim leading blank lines
-            while len(cell_lines) > 0 and len(cell_lines[0]) == 0:
-                cell_lines.pop(0)
+            new_cell_lines = cell_lines
+            while len(new_cell_lines) > 0 and len(new_cell_lines[0]) == 0:
+                new_cell_lines.pop(0)
+                start += 1
+
             # trim trailing blank lines
-            while len(cell_lines) > 0 and len(cell_lines[-1]) == 0:
-                cell_lines.pop(-1)
+            while len(new_cell_lines) > 0 and len(new_cell_lines[-1]) == 0:
+                new_cell_lines.pop(-1)
```

Comment:
Shouldn't it affect some offset as well? Otherwise a cell in the middle of the notebook with 7 trailing blank lines would produce incorrect start_line results, resulting in corruption of migration.

Add a unit test for this

Reply:
IN fact the existing tests all have trailing blank lines in cells and so this is already tested